src/Uno.UI/UI/LayoutHelper.cs

```diff
@@ -106,6 +107,8 @@ internal static (Point offset, bool overflow) GetAlignmentOffset(this IFramework
 					offset.X = 0;
 					overflow = true;
 					break;
+				case HorizontalAlignment.Stretch when e is Popup && clientSize.Width == 0:
```

Comment:
Unless this is really what UWP is doing, it'd be better to put Popup-specific workarounds in Popup-specific code if possible, rather than the general layouting logic.